internal/provider/group_resource_test.go

```diff
@@ -397,14 +397,12 @@ func testAccCheckGroupDestroy(s *terraform.State) error {
 				ID: rs.Primary.ID,
 			},
 		)
-		if err == nil {
-			if group != nil {
-				return fmt.Errorf("Group %s still exists", rs.Primary.ID)
-			}
+		if err != nil {
```

Comment:
Now that the OAPI spec states that a 404 is an error code we support it seems the SDK doesn't return it as an unhandled error but within the returned value so we need to update the logic for how we determine if it's a404